lib/theme_check/language_server/handler.rb

```diff
@@ -98,8 +100,10 @@ def text_document_uri(params)
         path_from_uri(params.dig('textDocument', 'uri'))
       end
 
-      def path_from_uri(uri)
-        uri&.sub('file://', '')&.sub('/c%3A', '')
+      def path_from_uri(uri_string)
+        return if uri_string.nil?
+        uri = URI(uri_string)
+        CGI.unescape(uri.path)
```

Comment:
This has the added bonus of handling windows paths for us (`C%3a` becomes `c:` as it should).